src/test/java/org/openmrs/module/mirebalais/smoke/SimpleFormTest.java

```diff
@@ -0,0 +1,78 @@
+package org.openmrs.module.mirebalais.smoke;
+
+import org.hamcrest.Matchers;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+public abstract class SimpleFormTest extends DbTest {
+
+    public abstract void testForms() throws Exception;
+
+    /**
+     * Checks that every section of a multi-section form loads without an error. Does not input data or submit.
+     *
+     * Precondition: must have the PIH Visit page open with no encounters
+     *
+     * @param formLabel The text of the action button that opens the form
+     * @throws Exception
+     */
+    public void testSectionedFormLoads(String formLabel) throws Exception {
+        visitNote.openForm(By.linkText(formLabel));
+        int numSections = driver.findElements(By.className("visit-element")).size();
+        for (int i = 0; i < numSections; i++) {
+            WebElement sectionHeader = driver.findElements(By.className("visit-element")).get(i);
+            WebElement editButton = getVisible(sectionHeader.findElements(By.className("edit-encounter")));
+            if (editButton != null) {
+                editButton.click();
+                assertThat(driver.findElement(By.id("htmlform")), Matchers.notNullValue());
+                driver.navigate().back();
+            } else {
+                fail("No edit button was visible for form section " + sectionHeader.getText());
+            }
+        }
+        deleteTheEncounter();
+    }
+
+    public void testSimpleFormLoads(String formLabel) throws Exception {
+        visitNote.openForm(By.linkText(formLabel));
+        assertThat(driver.findElement(By.id("htmlform")), Matchers.notNullValue());
+        driver.navigate().back();
+    }
+
+    /**
+     * Deletes the visible encounter. There should only be one.
+     */
+    private void deleteTheEncounter() throws Exception {
```

Comment:
Are we doing this primarly for data cleanup after the test?

I *think* if you are using one of the "test patients" (ie adultTestPatient) the DB test cleanup will remove any encounters for that patient?

Oh, actually I see now, this is probably because a single test tries to open all the forms, and so you want to clean this up mid-test?  Makes sense!

Reply:
> this is probably because a single test tries to open all the forms, and so you want to clean this up mid-test?

Yep!